x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/config.ts

```diff
@@ -93,13 +106,13 @@ export function getTopMetricsAggConfig(
         this.aggConfig.sortSettings = rest;
 
         if (isValidSortDirection(order)) {
-          this.aggConfig.sortSettings.order = order;
+          this.aggConfig.sortSettings!.order = order;
```

Comment:
Are these non-null assertions necessary? Looks like `this.aggConfig.sortSettings` will be `{}` even if there's nothing defined in `rest`.

Reply:
TS is having a hard time inferring it. Didn't manage to find out what the root cause is.

Reply:
Maybe we can add a comment why the `!` is necessary. Or would having an additional type guard like `isSortSettings()` solve it?

Reply:
Looks like my IDE was acting weird, removed in https://github.com/elastic/kibana/pull/138821/commits/01a398d73885c5e81255efb799f775ab216ecd57